build.rst: Update environment variable docs

stuarteberg opened the pull request.

asmeurer commented on docs/source/build.rst:
```diff
@@ -329,7 +329,10 @@ Environment variables set during the build process
+Windows (``bld.bat``) during the build process.
+(By default, these are the only variables available to your build script --
+no variables are inherited from the environment in which you invoked ``conda build``.
+See :ref:`inherited-env-vars` to override this behavior.)
```
This is not entirely true. Some variables, like `LANG` are inherited (fixed in master), and some like `PATH` are inherited and modified. Probably this just needs to be updated here, though.

stuarteberg commented:
OK, I added the missing environment variables to the table and tweaked the language to be more accurate.  I also corrected `MACOSX_DEPLOYMENT_TARGET` while I was at it.
